openrr-base/src/base_odometry.rs

```diff
@@ -0,0 +1,66 @@
+use arci::{BaseVelocity, Isometry2, Vector2};
+use parking_lot::Mutex;
+
+#[derive(Debug)]
+pub struct BaseOdometry {
+    position: Mutex<Isometry2<f64>>,
+    last_update_timestamp: Mutex<std::time::Instant>,
+}
+
+impl BaseOdometry {
+    pub fn new() -> Self {
+        Self {
+            position: Default::default(),
+            last_update_timestamp: Mutex::new(std::time::Instant::now()),
+        }
+    }
+
+    pub fn translate(&self, velocity: &BaseVelocity) {
+        let mut pose = self.position.lock();
+        let mut update_time = self.last_update_timestamp.lock();
+
+        let x = pose.to_owned().translation.x;
+        let y = pose.to_owned().translation.y;
+        let theta = pose.to_owned().rotation.angle();
+        let delta = update_time.to_owned().elapsed().as_secs_f64();
+
+        *pose = Isometry2::new(
+            Vector2::new(
+                x + (velocity.x * theta.cos() - velocity.y * theta.sin()) * delta,
+                y + (velocity.x * theta.sin() + velocity.y * theta.cos()) * delta,
+            ),
+            theta + velocity.theta * delta,
+        );
```

Comment:
I feel better to use Isometry2::mut.
I mean define delta_pose as Isometry2, then you can do

```rust
*pose *= delta_pose;
```

(I don't know about lock, sorry.)